@xen-orchestra/lite/src/directives/tooltip.directive.ts

```diff
@@ -0,0 +1,64 @@
+import { isString } from "lodash-es";
+import type { Options } from "placement.js";
+import type { DirectiveBinding } from "vue";
+import { useEventListener } from "@vueuse/core";
+import { useTooltipStore } from "@/stores/tooltip.store";
+
+export type TooltipOptions =
+  | string
+  | {
+      content: string;
+      placement: Options["placement"];
+      disabled?: boolean | ((target: HTMLElement) => boolean);
+    };
+
+export type TooltipProps = {
+  content: string;
+  placement: Options["placement"];
+  disabled?: boolean | ((target: HTMLElement) => boolean);
+  target: HTMLElement;
+};
+
+const optionsToProps = (
+  target: HTMLElement,
+  userOptions: TooltipOptions
+): TooltipProps => {
+  if (isString(userOptions)) {
+    return {
+      content: userOptions,
+      disabled: false,
+      placement: "top",
+      target,
+    };
+  }
+
+  return {
+    ...userOptions,
+    target,
+  };
+};
+
+export default function vTooltip(
+  target: HTMLElement,
+  binding: DirectiveBinding<TooltipOptions>
+) {
+  const tooltipStore = useTooltipStore();
+  const props = optionsToProps(target, binding.value);
```

Comment:
`props` should be a computed no ? What's happend if `binding.value` change ?

Reply:
No need for a computed here.

Function directive (`const vDirective = () => { ... }`) is a shorthand for:

```
const func = () => { ... }

const vDirective= {
  mounted: func,
  updated: func,
}
```

So the function is re-executed each time the binding is changing.